lib/util/http/scm-manager.ts

```diff
@@ -0,0 +1,193 @@
+      {
+        scmmContentType: CONTENT_TYPES.REPOSITORY,
+      },
+      RepoSchema,
+    );
+    return response.body;
+  }
+
+  public async getAllRepos(): Promise<Repo[]> {
+    const response = await this.getJson(
+      URLS.ALL_REPOS,
+      {
+        scmmContentType: CONTENT_TYPES.REPOSITORIES,
+      },
+      PagedRepoSchema,
+    );
+
+    return response.body._embedded.repositories;
+  }
+
+  public async getDefaultBranch(repo: Repo): Promise<string> {
+    const defaultBranchUrl = repo._links['defaultBranch'] as Link;
+    const response = await this.getJson(
+      defaultBranchUrl.href,
+      {
+        scmmContentType: CONTENT_TYPES.GIT_CONFIG,
+      },
+      DefaultBranchSchema,
+    );
+
+    return response.body.defaultBranch;
+  }
+
+  public async getAllRepoPrs(repoPath: string): Promise<PullRequest[]> {
+    const response = await this.getJson(
+      URLS.PULLREQUESTS_WITH_PAGINATION(repoPath),
+      {
+        scmmContentType: CONTENT_TYPES.PULLREQUESTS,
+      },
+      PagedPullRequestSchema,
+    );
+    return response.body._embedded.pullRequests;
+  }
+
+  public async getRepoPr(repoPath: string, id: number): Promise<PullRequest> {
+    const response = await this.getJson(
+      URLS.PULLREQUEST_BY_ID(repoPath, id),
+      {
+        scmmContentType: CONTENT_TYPES.PULLREQUEST,
+      },
+      PullRequestSchema,
+    );
+
+    return response.body;
+  }
+
+  public async createPr(
+    repoPath: string,
+    params: PullRequestCreateParams,
+  ): Promise<PullRequest> {
+    const createPrResponse = await this.postJson(URLS.PULLREQUESTS(repoPath), {
+      scmmContentType: CONTENT_TYPES.PULLREQUEST,
+      body: params,
+      headers: {
+        'Content-Type': CONTENT_TYPES.PULLREQUEST,
+      },
+    });
+
+    const getCreatedPrResponse = await this.getJson(
+      /* istanbul ignore next: Just to please the compiler, location would never be undefined */
+      createPrResponse.headers.location ?? '',
+      {
+        scmmContentType: CONTENT_TYPES.PULLREQUEST,
+      },
+      PullRequestSchema,
+    );
+
+    return getCreatedPrResponse.body;
+  }
+
+  public async updatePr(
+    repoPath: string,
+    id: number,
+    params: PullRequestUpdateParams,
+  ): Promise<void> {
+    const currentPr = await this.getRepoPr(repoPath, id);
+    await this.putJson(URLS.PULLREQUEST_BY_ID(repoPath, id), {
+      body: this.mergePullRequestWithUpdate(currentPr, params),
+      headers: {
+        'Content-Type': CONTENT_TYPES.PULLREQUEST,
+      },
+    });
+  }
+
+  private mergePullRequestWithUpdate(
+    pr: PullRequest,
+    updateParams: PullRequestUpdateParams,
+  ): PullRequest {
+    return {
+      ...pr,
+      title: updateParams.title,
+      description: updateParams.description
+        ? updateParams.description
+        : pr.description,
+      status: updateParams.status ? updateParams.status : pr.status,
+      target: updateParams.target ? updateParams.target : pr.target,
+    };
+  }
```

Comment:
move to seperate class in platform/scm-manager path.
configure endpoint like here:

https://github.com/renovatebot/renovate/blob/4032f206876960f36acfdce28982ac9167a48aa0/lib/util/http/gitea.ts#L6-L9